src/Microsoft.AspNetCore.SignalR.Core/HubLifetimeManager.cs

```diff
@@ -2,37 +2,40 @@
         public abstract Task OnDisconnectedAsync(HubConnectionContext connection);
 
-        public abstract Task SendAllAsync(string methodName, object[] args);
+        public abstract Task SendAllAsync(string methodName, object[] args, CancellationToken cancellationToken = default);
 
-        public abstract Task SendAllExceptAsync(string methodName, object[] args, IReadOnlyList<string> excludedConnectionIds);
+        public abstract Task SendAllExceptAsync(string methodName, object[] args, IReadOnlyList<string> excludedConnectionIds, CancellationToken cancellationToken = default);
 
-        public abstract Task SendConnectionAsync(string connectionId, string methodName, object[] args);
+        public abstract Task SendConnectionAsync(string connectionId, string methodName, object[] args, CancellationToken cancellationToken = default);
 
-        public abstract Task SendConnectionsAsync(IReadOnlyList<string> connectionIds, string methodName, object[] args);
+        public abstract Task SendConnectionsAsync(IReadOnlyList<string> connectionIds, string methodName, object[] args, CancellationToken cancellationToken = default);
 
-        public abstract Task SendGroupAsync(string groupName, string methodName, object[] args);
+        public abstract Task SendGroupAsync(string groupName, string methodName, object[] args, CancellationToken cancellationToken = default);
 
-        public abstract Task SendGroupsAsync(IReadOnlyList<string> groupNames, string methodName, object[] args);
+        public abstract Task SendGroupsAsync(IReadOnlyList<string> groupNames, string methodName, object[] args, CancellationToken cancellationToken = default);
 
-        public abstract Task SendGroupExceptAsync(string groupName, string methodName, object[] args, IReadOnlyList<string> excludedConnectionIds);
+        public abstract Task SendGroupExceptAsync(string groupName, string methodName, object[] args, IReadOnlyList<string> excludedConnectionIds, CancellationToken cancellationToken = default);
 
-        public abstract Task SendUserAsync(string userId, string methodName, object[] args);
+        public abstract Task SendUserAsync(string userId, string methodName, object[] args, CancellationToken cancellationToken = default);
 
-        public abstract Task SendUsersAsync(IReadOnlyList<string> userIds, string methodName, object[] args);
+        public abstract Task SendUsersAsync(IReadOnlyList<string> userIds, string methodName, object[] args, CancellationToken cancellationToken = default);
 
-        public abstract Task AddToGroupAsync(string connectionId, string groupName);
+        public abstract Task AddToGroupAsync(string connectionId, string groupName, CancellationToken cancellationToken = default);
 
-        public abstract Task RemoveFromGroupAsync(string connectionId, string groupName);
+        public abstract Task RemoveFromGroupAsync(string connectionId, string groupName, CancellationToken cancellationToken = default);
```

Comment:
Since these are all called by methods that just got CTs (`IClientProxy.Send` and `IGroupManager.AddToGroupAsync/RemoveFromGroupAsync`) the lifetime manager should get to observe cancellation if it wants to.